In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats

pd.options.display.float_format = "{:.2f}".format

In [2]:
# Read case data
df = pd.read_excel("4. Case 3 - Two-sample t-test.xlsx")

In [3]:
df.head()

Store Id      Display Type  Sales Volume
0   3846186  Old_display_type       2038.31
1   1083410  Old_display_type       2017.29
2   4278951  Old_display_type       1746.47
3   6670048  Old_display_type       2061.78
4   5054220  Old_display_type       2253.76

In [4]:
# Create arrays with units sold for 'Old_display_type' and 'New_display_type'
data_o = df[df["Display Type"] =="Old_display_type"]["Sales Volume"].values
data_n = df[df["Display Type"] =="New_display_type"]["Sales Volume"].values

In [5]:
# Mean sales for old and new display types
print("Mean Old: " + str(np.mean(data_o)))
print("Mean New: " + str(np.mean(data_n)))

Mean Old: 2002.4714000000001
Mean New: 2099.6049999999996


The sales volume with the new display (2099.60) is higher than that of the old display type (2002.47). But is this difference statistically significant? We aim to answer this question with a statistical test.

Before proceeding with the test, let's clearly define our null and alternative hypotheses:

- H0: µ(Sales_VolNew) - µ(Sales_VolOld) = 0
- H1: µ(Sales_VolNew) - µ(Sales_VolOld) > 0

Additionally, we'll set the significance level, alpha, to 0.05.

The two-sample t-test (independent t-test) compares the means of two independent samples to determine if there's a significant difference between them. In Python, we use the **scipy.stats.ttest_ind function()** to perform this test.

The following is a breakdown of the main parameters for the **scipy.stats.ttest_ind()** function.

1.	 **a, b**: These are the two input arrays that contain the sample data for comparison. 'a' represents the first sample, and 'b' represents the second.
2.	**equal_var (default = True)**: This parameter specifies whether the variances of the two samples are considered equal. When True, the test uses a pooled estimate of variance. If False, it uses separate variance estimates for each sample, known as Welch's t-test.
3.	**alternative**: This parameter defines the alternative hypothesis with the following options: 
*	'two-sided': Indicates the means of the two distributions are unequal (default)
*	'less': The mean of the first sample (a) is less than that of the second sample (b).
*	'greater': The mean of the first sample (a) is greater than that of the second sample (b).

In our case, we choose 'greater' to align with our alternative hypothesis that the mean sales volume for the new display is greater than that for the old display. 

Since we've established that the sales volume variances for both displays are equal, we'll set **equal_var=True**.

In [6]:
# Run the two-sample t-test
stats.ttest_ind(data_n, data_o, equal_var=True, alternative = "greater")

TtestResult(statistic=1.7693132318181242, pvalue=0.039976188061622175, df=98.0)

As the p-value (0.0400) is below the significance level of 0.05, we can reject the null hypothesis supporting the alternative.

The sales volume with the new display is higher than that of the old display type (2099.60 vs. 2002.47), but we've also established that this difference is statistically significant.

For the two-sample t-test, the formula to determine degrees of freedom is:

df = N1 + N2 – 2.

where:
- df = degrees of freedom
- N1 = sample size of the first group
- N2 = sample size of the first group

Therefore, in this case, df = 98